packages/theme-check-common/src/checks/valid-block-preset-settings/index.ts

```diff
@@ -0,0 +1,115 @@
+import { basename } from '../../path';
+import { LiquidCheckDefinition, Severity, SourceCodeType, ThemeBlock } from '../../types';
+import { Preset } from '../../types/schemas/preset';
+
+export const ValidBlockPresetSettings: LiquidCheckDefinition = {
+  meta: {
+    code: 'ValidBlockPresetSettings',
+    name: 'Reports invalid preset settings for a theme block',
+    docs: {
+      description: 'Reports invalid preset settings for a theme block',
+      recommended: true,
+      url: 'https://shopify.dev/docs/storefronts/themes/tools/theme-check/checks/preset-key-exists',
+    },
+    severity: Severity.ERROR,
+    type: SourceCodeType.LiquidHtml,
+    schema: {},
+    targets: [],
+  },
+
+  create(context) {
+    function getSchema() {
+      const name = basename(context.file.uri, '.liquid');
+      switch (true) {
+        case isBlock(context.file.uri):
+          return context.getBlockSchema?.(name);
+        case isSection(context.file.uri):
+          return context.getSectionSchema?.(name);
+        default:
+          return undefined;
+      }
+    }
+
+    function getInlineSettingsTypesandKeys(settings: ThemeBlock.Schema['settings']) {
+      if (!settings) return [];
+      return settings.map((setting: { id: any; type: any }) => ({
+        id: setting.id,
+        type: setting.type,
+      }));
+    }
+
+    function getPresetSettingsKeys(presets: Preset.Preset[]) {
+      const allKeys: string[] = [];
+      for (const preset of presets) {
+        if (preset.settings) {
+          allKeys.push(...Object.keys(preset.settings));
+        }
+      }
+      return allKeys;
+    }
+
+    function getPresetBlockSettingsKeys(blocks: Preset.PresetBlocks) {
+      const allKeys: string[] = [];
+      for (const block of Object.values(blocks)) {
+        for (const [_, blockData] of Object.entries(block)) {
+          if (blockData && typeof blockData === 'object' && 'settings' in blockData) {
+            const settings = blockData.settings;
+            if (settings && typeof settings === 'object') {
+              allKeys.push(...Object.keys(settings));
+            }
+          }
+        }
+      }
+      return allKeys;
+    }
+
+    return {
+      async LiquidRawTag(node) {
+        if (node.name !== 'schema' || node.body.kind !== 'json') {
+          return;
+        }
+
+        const schema = await getSchema();
+        if (!schema) return;
+        if (schema.validSchema instanceof Error) return;
+
+        const validSchema = schema.validSchema;
+        const settingsKeys = getInlineSettingsTypesandKeys(validSchema.settings);
+        const presetSettingsKeys = getPresetSettingsKeys(validSchema.presets ?? []);
+
+        for (const key of presetSettingsKeys) {
+          if (!settingsKeys.some((setting) => setting.id === key)) {
+            context.report({
+              message: `Preset setting "${key}" does not exist in the block's settings`,
+              startIndex: 0,
+              endIndex: 0,
```

Comment:
We need to highlight the node. I don't think it's ok to have 0:0 for this check. 

`schema` has a `ast` property you can use to find the node. We also have the `nodeAtPath` util that can give you the AST node for a given "JSONPath"

```ts
const schema = await getSchema();
if (!schema) return;
if (isError(schema.validSchema) || isError(schema.ast)) return;

// find out that ["presets", 0, "settings", "unknown-id"] is invalid:
const settingsNode = nodeAtPath(schema.ast, ["presets", 0, "settings"]) as ObjectNode;
if (!settingsNode) {} // something went wrong

const unknownIdNode = settingsNode.
```